src/buffer/channel.rs

```diff
@@ -203,25 +203,42 @@ pub fn view<'a>(
                             .into(),
                         )
                     }
-                    message::Source::Action => {
+                    message::Source::Action(user) => {
                         let marker = message_marker(max_nick_width, theme::selectable_text::action);
 
-                        let message = message_content(
+                        let message_content = message_content(
                             &message.content,
                             theme,
                             scroll_view::Message::Link,
                             theme::selectable_text::action,
                             config,
                         );
 
+                        let text_container =
+                            container(message_content).style(move |theme| match user {
+                                Some(user) => match our_nick {
+                                    Some(nick)
+                                        if message::references_user(
+                                            user.nickname(),
+                                            nick,
+                                            message,
+                                        ) =>
+                                    {
+                                        theme::container::highlight(theme)
+                                    }
+                                    _ => Default::default(),
+                                },
+                                _ => Default::default(),
+                            });
+
```

Comment:
This could be simplified a bit, not tested but something like:

```rust
let text_container = container(message_content).style(|theme| {
    if let (Some(user), Some(nick)) = (user, our_nick) {
        if message::references_user(user.nickname(), nick, message) {
            return theme::container::highlight(theme);
        }
    }
    Default::default()
});
```

Reply:
Ah! That's so much nicer. I didn't know that you could combine the `Some(..)` assignment that way. I did think that there was something to optimize there! Thanks for this :pray: